Don't pre-zero freshly allocated memory.

chrisvest opened the pull request:
This is safe because it will always be overwritten or zeroed out when we page fault.

This _might_ improve performance on high-mem and NUMA systems. Needs testing.

chrisvest committed:
Don't pre-zero freshly allocated memory.

This is safe because it will always be overwritten or zeroed out when we page fault.

chrisvest committed:
Add a hidden configuration for the MemoryManager grab size

thobe commented on community/unsafe/src/main/java/org/neo4j/unsafe/impl/internal/dragons/MemoryManager.java:
```diff
@@ -29,7 +29,8 @@
 {
-    private static final long GRAB_SIZE = 32 * 1024 * 1024; // 32 MiB
+    private static final long GRAB_SIZE = Integer.getInteger(
+            "org.neo4j.unsafe.impl.internal.dragons.MemoryManager.grabSize", 32 * 1024 * 1024 ); // 32 MiB
```
could we replace this with `MemoryManager.class.getName() + ".GRAB_SIZE"`?

chrisvest commented:
NUMA testing done, and we've adversarially verified that the pre-zeroing indeed isn't necessary, so I think it's ready for merge.

chrisvest committed:
Fix page cache memory reporting when `dump_configuration` is enabled

The previous code did integer multiplication of the number of pages in the cache, and the page size, which would overflow if the page cache was given more than 2 GiBs of memory.
This does not change how the page cache memory allocation is calculated, which was already correct, it only changes how this allocation is reported by `dump_configuration`.
